### DISCUSSION TITLE
초기 역량 관리(저장) 방법

### DISCUSSION THREAD

**Discussion by Hyeon9mak** [woowacourse/prolog#559] [Feature]
현재 백엔드/프론트엔드 초기역량 파일을 백엔드 서버측에서 CSV파일로 관리하고 있었습니다!
초기 역량 데이터에 변화가 생겼을 때 (초기 역량을 추가한다던지, 뺀다던지)
재빌드/재배포 해야하는 상황을 만들고 싶지 않아서 고민하다가 CSV 파일로 초기역량 관리를 할 생각을 했습니다!

<br>

그러나 dev 서버 배포 이후 계속해서 `NullPointerException`을 마주하고 있습니다.
처음에는 파일의 경로를 못찾는게 아닐까 했는데, 엘라스틱 빈즈토크에서 application.jar 파일 내부를 확인해보면 결과 CSV 파일들은 잘 저장되어 있었고
`FileNotFoundException`을 따로 catch하도록 해봤으나 이를 지나치고 `NullPointerException`이 계속 터지고 있었습니다!
또 ‘CSV 파일에서 문자열을 나누는 기준인 `\\|` 가 빈즈토크에선 다르게 해석되지 않을까?’ 라고도 접근해서
`\\|` 를 `---`로 바꿔보았는데도 계속해서 `NullPointerException`을 마주하고 있습니다….

<br>

더 이상 해결 방법도 떠오르지 않고... 엘라스틱 빈즈토크에도 빌드 후 압축된 jar 파일만 전달 되기 때문에

> 초기 역량 데이터에 변화가 생겼을 때 (초기 역량을 추가한다던지, 뺀다던지) 재빌드/재배포 해야하는 상황을 만들고 싶지 않아서

라는 장점이 퇴색된 상태입니다.
그래서 다른 저장방법을 고민해봐야할까 싶은데, 어디에 어떻게 저장할지 고민이 됩니다!!

**Comment by sihyung92** [comment]
npe를 주깁시다 npe는 나의 원수

---

`더 이상 해결 방법도 떠오르지 않고... 엘라스틱 빈즈토크에도 빌드 후 압축된 jar 파일만 전달 되기 때문에
초기 역량 데이터에 변화가 생겼을 때 (초기 역량을 추가한다던지, 뺀다던지) 재빌드/재배포 해야하는 상황을 만들고 싶지 않아서
라는 장점이 퇴색된 상태입니다.`
라고 한다면 dataInitializer, 어플리케이션 부팅단에서 삽입하시죠.. :) 중복 상황처리 해놓고

csv관련 로직은 나중에 관리자도구를 집어 넣으면서 활용하면 어떨까 생각합니다

**Comment by bperhaps** [comment]
csv를 읽지 못하는 문제는 저번에 말했듯, 압축 해제와 관련된 문제이고, 나는 db에 넣는게 괜찮아 보이는데......

**Reply by Hyeon9mak** [reply]
압축해제도 너잘 블로그 참고해서 해봤는데.. ㅠㅠ

**Comment by Hyeon9mak** [comment]
으아?! 디비?! 데이터로더?!
디비 꼬?!

**Reply by sihyung92** [reply]
하지만 넌 권한없다 db 쿼리 에디터
